packages/store/src/operators/of-action.ts

```diff
@@ -74,6 +76,32 @@ export function ofActionErrored(...allowedTypes: any[]) {
   return ofActionOperator(allowedTypes, [ActionStatus.Errored]);
 }
 
+/**
+ * RxJS operator for selecting out specific actions.
+ *
+ * This will grab an action and return true when dispatched and false when completed
+ * When used with multiple actions, will return true when ALL actions has been dispatched or only some of them has completed,
+ * and false when ALL are completed
+ */
+export function ofActionExecuting(...allowedTypes: ActionType[]) {
+  return (source: Observable<ActionContext>) => {
+    const executionTypes: Observable<boolean>[] = allowedTypes.map((type: ActionType) =>
+      source.pipe(mapToActionExecuting(type))
+    );
+
+    return combineLatest(...executionTypes).pipe(
+      map((actionsExecuting: boolean[]) => {
+        const score: number = actionsExecuting.reduce(
+          (acc: number, status: boolean) => acc + Number(status),
+          0
+        );
+
+        return score > 0;
+      })
+    );
+  };
```

Comment:
I think that everything in this operator can be made a lot simpler through the use of the `scan` rxjs operator.
Something like this (just a brain dump, may not work):
```
return (source: Observable<ActionContext>) => {
  return source.pipe(
    ofAction(allowedTypes),
    scan((acc: number, { status }: ActionContext) => {
      if (status === ActionStatus.Dispatched) {
        return acc + 1;
      }
      if (status === ActionStatus.Successful 
          || status === ActionStatus.Canceled
          || status === ActionStatus.Errored
        ) {
        return acc - 1;
      }
      return acc;
    }, 0),
    map( ( numberExecuting: number) => numberExecuting > 0 ),
    distinctUntilChanged(),
  );
}
```

Reply:
had to make a couple of changes to try this approach, but existing test dont pass, specifically those that test multiple actions (not the same) are dispatched, the idea of the operator is to return true when all of them are done hence the combineLatest

Reply:
Is the issue you are having here related to the `distinctUntilChanged()`?
If you remove that do your tests work? (the valid ones... see my review below)

Reply:
removing `distinctUntilChanged()` fixes the corrected tests of same action being dispatched multiple times but fails when you dispatch multiple different actions. The expectation there is that you would start receving `true` or  `false` once all the actions have been dispactched hence the combineLatest, do you think that behaviour is not correct?

Reply:
btw the complete code using your approach looks like this:
```
export function ofActionExecuting(...allowedTypes: ActionType[]) {
  return (source: Observable<ActionContext>) => {
    return source.pipe(
      ofActionOperator(allowedTypes, [
        ActionStatus.Dispatched,
        ActionStatus.Successful,
        ActionStatus.Canceled,
        ActionStatus.Errored
      ], () => {
        return map(ctx => ctx);
      }),
      scan((acc: number, ctx: ActionContext) => {
        if (ctx.status === ActionStatus.Dispatched) {
          return acc + 1;
        }
        if (ctx.status === ActionStatus.Successful
          || ctx.status === ActionStatus.Canceled
          || ctx.status === ActionStatus.Errored
        ) {
          return acc - 1;
        }
        return acc;
      }, 0),
      map((numberExecuting: number) => numberExecuting > 0)
    );
  };
```
looks lilke ofAction, uses a map function that just returns the payload of the action and not the status